File: Programmers/Python/Level 2/가장 큰 수/solution.py
# https://programmers.co.kr/learn/courses/30/lessons/42746

def solution(numbers):
    # 숫자의 자릿수로 정렬을 하기위해 원소들을 문자로 바꿈
    numbers = list(map(str, numbers))
    # 원소의 최대값이 1000이기때문에 자릿수를 맞춰주기 위해 x * 4으로 key를 줌
    numbers.sort(key=lambda x: x * 4, reverse=True)
    answer = str(int("".join(numbers)))  # 0이 들어왔을때 0이 중복되는것을 막기 위해 int로 변환
    return answer

File: Programmers/Javascript/Level 2/가장 큰 수/solution.js
// https://school.programmers.co.kr/learn/courses/30/lessons/42746?language=javascript

function solution(numbers) {
  numbers = numbers
    .map((value) => {
      // 자릿수로 비교를하기위해 문자열로 변환
      return String(value);
    })
    .sort((a, b) => {
      // 합친 숫자값을 크기순으로 정렬하기위해 단항 연산자로 정수로 바꾼다음 비교
      return +(b + a) - +(a + b);
    });

  const result = numbers.join("");
  // "000"같은 경우를 방지하기위해 정수로 변환하여 검사
  return +result === 0 ? "0" : result;
}
